src/components/user-profile/index.js

```diff
@@ -1,30 +1,23 @@
 import { useState, useEffect } from 'react';
 import { userContext } from '@store/user/user-context';
 import classNames from '@components/user-profile/user-profile.module.scss';
-import { getUserSelf } from '../../helper-functions/action-handlers';
 
 const UserProfile = () => {
   const [isLoading, setIsLoading] = useState(true);
   const {
     user,
-    setUser,
     isSuperUser,
-    setIsSuperUser,
     isSuperUserMode,
     setIsSuperUserMode,
+    makeApiCallToVerifySuperUser,
   } = userContext();
 
   useEffect(() => {
-    async function getUserProfile() {
-      const { data } = await getUserSelf();
+    (async () => {
+      await makeApiCallToVerifySuperUser();
       setIsLoading(false);
-      setUser(data);
-      if (data) {
-        setIsSuperUser(data.roles.super_user);
-      }
-    }
-    getUserProfile();
-  }, [setUser, setIsSuperUser]);
+    })();
+  }, [makeApiCallToVerifySuperUser]);
```

Comment:
API Call in useEffect :O

Reply:
I need to read more about this. I will create a new issue and work on this part. Resolving this so that it can be shipped to production and used.